app.js

```diff
@@ -1,227 +1,47 @@
-    // manipulate data received from submission
-    const { data = {}, errors = {} } = bdayInteractive({ type, values })
-    if (Object.values(errors).length) {
-      return res.status(200).json({ response_action: 'errors', errors })
-    }
-    const {
-      ref,
-      response_url,
-      command,
-      channel_id,
-    } = JSON.parse(private_metadata)
-
-    const payload = {
-      ref,
-      command,
-      type,
-      view_id: id,
-      hash,
-      response_url,
-      data,
-      blocks,
-      action: actions[0] || [],
-      channel_id,
-      sender
-    }
-
-    const { worker } = routes.bday
-
-    if (DEPLOYED) {
-      lambda.invoke({
-        FunctionName: getFuncName('slack'),
-        InvocationType: 'Event',
-        Payload: JSON.stringify({ type: 'bday', payload }),
-      }, (err) => {
-        if (err) {
-          console.error(err)
-          return res.status(200).json({ response_type: 'ephemeral', text: 'Failed to process bday command' })
-        }
-        if (type === 'view_submission') {
-          return res.status(200).json({ 'response_action': 'clear' })
-        }
-        return res.sendStatus(200)
-      })
-    } else {
-      worker(payload).catch(console.error)
-      if (type === 'view_submission') {
-        // needs to have only <response_action> for modals submission
-        return res.status(200).json({ 'response_action': 'clear' })
-      }
-      // modal interactions need to have acknowledgement
-      return res.sendStatus(200)
-    }
+/* sample command object given to listener({ command, ack, ... })
+  {
+    token: 'm4Q8xY7jIYRbwM0W7u7utV30',
+    team_id: 'T1FDR4NG3',
+    team_domain: 'eqworks',
+    channel_id: 'CCCB6QD9S',
+    channel_name: 'bot-cmd', // <- this eliminates the need for getChannelName
+    user_id: 'U1FF6KSF9',
+    user_name: 'woozyking',
+    command: '/diff',
+    text: 'snoke',
+    api_app_id: 'A6HPM5VC0',
+    is_enterprise_install: 'false',
+    response_url: 'https://hooks.slack.com/commands/T1FDR4NG3/2956432472786/J8Q17JPFukiLurI8zbVJzTCX',
+    trigger_id: '2956462218819.49467158547.0dae80ec6d595b045efcf7a8fee079e7'
   }
+*/
+Object.entries(commands).forEach(([name, { listener }]) => {
+  app.command(`/${name}`, listener)
 })
 
-// catch-all error handler
-// eslint disable otherwise not able to catch errors
-// eslint-disable-next-line no-unused-vars
-app.use((err, req, res, next) => {
-  let { logLevel, statusCode } = err
-  const { message } = err
-  logLevel = logLevel || 'ERROR'
-  statusCode = statusCode || 500
-  // app log
-  // eslint-disable-next-line no-console
-  console.log(`[${logLevel}] - ${statusCode} - ${message}`)
-  if (logLevel === 'ERROR') {
-    console.error(`[ERROR] ${message}`, err.stack || err)
+// interactive views, by callback_id
+// TODO: ones that require rework are commented out
+Object.entries(commands).forEach(([name, { viewHandler }]) => {
+  if (viewHandler) {
+    app.view(name,  viewHandler)
```

Comment:
any reason not to combine the `forEach`s?
```js
Object.entries(commands).forEach(([name, { listener, viewHandler }]) => {
  app.command(`/${name}`, listener)
  if (viewHandler) {
    app.view(name,  viewHandler)
  }
}  
```

Reply:
probably an afterthought. no good reason and they should be combined